compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -673,7 +678,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
                 let mut restrict_type_params = false;
                 let mut unsatisfied_bounds = false;
-                if !unsatisfied_predicates.is_empty() {
+                let mut fallback_span = true;
+                if item_name.name == sym::count && self.is_slice_ty(actual, span) {
+                    let msg = "consider using `len` instead";
+                    if let SelfSource::MethodCall(expr) = source {
+                        let call_expr =
+                            self.tcx.hir().expect_expr(self.tcx.hir().get_parent_node(expr.hir_id));
+                        if let Some(span) = call_expr.span.trim_start(expr.span) {
+                            err.span_suggestion(
```

Comment:
```suggestion
                            err.span_suggestion_short(
```

This should change the output like so:

```diff
-help: consider using `len` instead: `.len()`
+help: consider using `len` instead
```

but the suggested code will still be displayed if it's in its own diagnostic window.